### PR TITLE
refactor(manager): pass through `UpdateArtifact` generic type

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes

Change the type of `UpdateArtifact` to pass through a generic type to `updatedDeps`

## Context

Discovered this as part of some other feature I am working on. 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/types.ts [hunk 1]
```diff
@@ -198,9 +198,9 @@ export interface UpdateArtifactsResult {
   file?: FileChange;
 }
 
-export interface UpdateArtifact {
+export interface UpdateArtifact<T = Record<string, any>> {
```
Comment (viceice) [inline lib/modules/manager/types.ts:201]: wouldn't like to see unknown instead of any in future

**Review by viceice — DISMISSED** [review]

File: lib/modules/manager/types.ts (hunk 1 above)
Comment (JamieMagee) [reply lib/modules/manager/types.ts:201]: I mainly chose `any` as it's currently the default type for `PackageDependency`'s generic type. I can change it here already if you'd like?

Comment (viceice) [reply lib/modules/manager/types.ts:201]: if it doesn't cause too many issues 🤗

**Review by viceice — COMMENTED** [review]
it seems we've one issue which needs to be fixed 🙂

**Review by rarkins — DISMISSED** [review]

**Comment by JamieMagee** [comment]
@viceice I added `FluxManagerData`

**Review by viceice — APPROVED** [review]

**Pull request merged**